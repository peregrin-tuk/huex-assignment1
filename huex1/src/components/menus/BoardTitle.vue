<template>
  <v-text-field
    :loading="loading"
    :value="title"
    color="success"
    @change="changeBoardName"/>
</template>

<script lang="ts">
  import Vue from 'vue'

  interface loading {
    loading: boolean | string
  }

  export default Vue.extend({
    name: 'BoardTitle',
    data: function (): loading {
      return {
        loading: false
      }
    },
    computed: {
      title() {
        if(!this.$store.state.currentBoard) return
        return this.$store.state.currentBoard.name
      }
    },
    methods: {
      changeBoardName: function (value: any) {
        if (typeof value !== 'string') return

        this.loading = 'secondary'
        this.$store.dispatch('updateBoardName', value).then(() => {
            this.loading = false
          }
        )
      }
    }
  })
</script>

<style lang="scss" scoped>
</style>
