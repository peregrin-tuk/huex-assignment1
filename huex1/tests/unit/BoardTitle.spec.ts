import {createLocalVue, shallowMount, mount} from '@vue/test-utils';
import BoardTitle from "@/components/menus/BoardTitle.vue";
// eslint-disable-next-line no-unused-vars
import Vuex, { Store } from 'vuex'
import Vuetify from 'vuetify';

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

describe('BoardTitle', () => {
  let actions: { updateBoardName: jest.Mock};
  let store: Store<any>;

  beforeEach(() => {
    actions = {
      updateBoardName: jest.fn()
    }
    store = new Vuex.Store({
      actions,
      state: {
        currentBoard: {
          name: 'Awesome name',
          key: '#123456789',
          content: [],
          id: '1234'
        }
      }
    })
  })

  it('should match snapshot', () => {
    const wrapper = shallowMount(BoardTitle, { 
      store,
      localVue,
    });

    expect(wrapper.html()).toMatchSnapshot();
  })

  it('shows current board title', async () => {
    const wrapper = mount(BoardTitle, { 
      store, 
      localVue,
    });

    const input = <HTMLInputElement>wrapper.find('.v-text-field__slot > input').element;

    expect(input.value).toBe(store.state.currentBoard.name);
  })

  // // Events get not correctly triggered on input
  // it('persists new board title after edit', () => {
  //   const wrapper = mount(BoardTitle, { 
  //     store, 
  //     localVue,
  //   });

  //   const input = wrapper.find('.v-text-field__slot > input');
  //   const inputElement =  <HTMLInputElement>input.element;
  //   inputElement.value = 'New board name';

  //   input.trigger('input')
  //   input.trigger('blur')

  //   expect(actions.updateBoardName).toHaveBeenCalledTimes(1)
  // })

})
