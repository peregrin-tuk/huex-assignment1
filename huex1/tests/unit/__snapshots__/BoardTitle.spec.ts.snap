// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardTitle should match snapshot 1`] = `<v-text-field-stub color="success" errorcount="1" errormessages="" messages="" rules="" successmessages="" value="Awesome name" backgroundcolor="" dense="true" hidedetails="true" label="Edit Title" loaderheight="2" clearicon="$clear" outlined="true" type="text" class="board-title"></v-text-field-stub>`;
